### ISSUE TITLE
Issue with onRippleCompletedListener and long click

### ISSUE THREAD

**Issue by LaurieScheepers** [traex/RippleEffect#67]
Hi, I'm setting a onRippleCompleted listener for a view in a recycler view and it works beautifully with a normal click.

The problem is that the view also has a onCreateContextMenuListener. And when the long click fires, showing the context menu, the ripple also completes.

Is there a way to cancel the ripple on a long click?

**Comment by LaurieScheepers** [comment]
Nevermind, found a workaround:

Add a long click listener to the view and in onLongClick, set the onRippleCompleteListener to null.

Also add a on click listener and set your onRippleCompleteListener there.

```
RippleView rippleView = (RippleView) view;
rippleView.setOnClickListener(this);
rippleView.setOnLongClickListener(this);
```

```
@Override
public void onClick(View v) {
        RippleView rippleView = (RippleView) v;
        rippleView.setOnRippleCompleteListener(mOnRippleCompleteListener);
}
```

```
@Override
public boolean onLongClick(View v) {
        // Cancel the ripple animation
        RippleView rippleView = (RippleView) v;

        rippleView.clearAnimation();
        rippleView.setOnRippleCompleteListener(null);
        return false;
}
```

The clear animation is probably not necessary.

This works great for me. The reason why I had to add the on ripple complete listener in the first place is because I saw when clicking on a RecyclerView item that has a ripple view as the root view, my activity is started before the animation finishes. I tried playing around with the ripple duration but no luck. I see some other people have posted the same bug.

But cool library, thanks!

**Comment by cosic** [comment]
@LaurieScheepers Look at [pull reques #54](https://github.com/traex/RippleEffect/pull/54). May be it's realization of your problem.

**Comment by DVassilev** [comment]
Is it possible to cancel the ripple effect for onClick and only have it for an onLongClick. Tried doing what you did but the other way around and it didn't seem to work.

**Comment by cosic** [comment]
@DVassilev I think it's possible.

**Comment by DVassilev** [comment]
Yeah I managed to get it working by setting ripple duration to 0 in onClick and to default value in onLongClick. However, the code inside onLongClick doesn't seem to be called every time but I think that might related to something else.

**Comment by JitenNumberTank** [comment]
Ripple View setOnLongClickListener called many times. Some times I get it twice call for onLongClick. Could anyone tell me where is problem in this?
